sql/src/main/java/io/crate/executor/transport/ExecutionPhasesTask.java

```diff
@@ -312,11 +313,13 @@ private void sendJobRequests(String localNodeId,
         }
     }
 
-    private List<PageDownstreamContext> getHandlerPageDownstreamContexts(Tuple<List<ExecutionSubContext>, List<ListenableFuture<Bucket>>> onHandler) {
-        final List<PageDownstreamContext> pageDownstreamContexts = new ArrayList<>(onHandler.v1().size());
-        for (ExecutionSubContext handlerExecutionSubContext : onHandler.v1()) {
-            if (handlerExecutionSubContext instanceof DownstreamExecutionSubContext) {
-                PageDownstreamContext pageDownstreamContext = ((DownstreamExecutionSubContext) handlerExecutionSubContext).pageDownstreamContext((byte) 0);
+    private List<PageDownstreamContext> getHandlerPageDownstreamContexts(JobExecutionContext.Builder builder,
+                                                                         List<Tuple<ExecutionPhase, RowReceiver>> handlerPhases) {
+        final List<PageDownstreamContext> pageDownstreamContexts = new ArrayList<>(handlerPhases.size());
+        for (Tuple<ExecutionPhase, RowReceiver> handlerPhase : handlerPhases) {
+            ExecutionSubContext ctx = builder.getSubContext(handlerPhase.v1().executionPhaseId());
```

Comment:
isn't it possible to use the JobExecutionContext here instead of the Builder? Then the builder doesn't require the new `getSubContext` method